lupa/tests/test.py

```diff
@@ -2620,6 +2622,115 @@ def test_nil_debug_traceback(self):
+
+        # FIXME: lupa._LuaObject.__eq__ might make this function simpler
+
+        obtained_args = lua_func(self.get_args)
+        self.assertEqual(len(obtained_args), len(args))
+        for a, b in zip(obtained_args, args):
+            self.assertEqualInLua(a, b)
+
+        obtained_kwargs = lua_func(self.get_kwargs)
+        self.assertEqual(len(obtained_kwargs), len(kwargs))
+        for key in kwargs:
+            self.assertEqualInLua(obtained_kwargs[key], kwargs[key])
+
+    def assertIncorrect(self, txt, error=TypeError, regex=''):
+        lua_func = self.lua.eval('function (f) return f(%s) end' % txt)
+        self.assertRaisesRegex(error, regex, lua_func, self.get_none)
+
+    def test_no_args(self):
+        self.assertResult('python.args{}', (), {})
+
+    def test_all_types(self):
+        # Positional arguments
+        args = self.lua.eval('''
+        {
+            42,
+            false,
+            "spam",
+            function() end,
+            coroutine.create(function() end),
+            {1, 2, 3},
+            python.none,
+        }
+        ''')
+        self.lua.globals()['args'] = args
+        self.assertResult('python.args(args)', tuple(args[i+1] for i in range(len(args))), {})
+
+        # Keyword arguments
+        kwargs = self.lua.table()
+        self.lua.globals()['kwargs'] = kwargs
+        self.lua.execute('''
+            for _, v in ipairs(args) do
+                kwargs[type(v)] = v
+            end
+        ''')
+        self.assertResult('python.args(kwargs)', (), dict(kwargs.items()))
+
+        # Invalid parameter to python.args
+        for objtype in kwargs:
+            if objtype != 'table':
+                self.assertIncorrect('python.args(kwargs["%s"])' % objtype,
+                        error=lupa.LuaError, regex="bad argument #1 to 'args'")
+        
+        # Invalid table keys
+        self.assertIncorrect('python.args{[0] = true}', error=IndexError, regex='table index out of range')
+        self.assertIncorrect('python.args{[2] = true}', error=IndexError, regex='table index out of range')
+        self.assertIncorrect('python.args{[3.14] = true}', regex='table key is neither an integer nor a string')
+        for objtype in kwargs:
+            if objtype not in {'number', 'string'}:
+                self.assertIncorrect('python.args{[kwargs["%s"]] = true}' % objtype,
+                        regex='table key is neither an integer nor a string')
+ 
+    def test_kwargs_merge(self):
+        self.assertResult('python.args{1, a=1}, python.args{2}, python.args{}, python.args{b=2}', (1, 2), dict(a=1, b=2))
+
+    def test_kwargs_merge_conflict(self):
+        self.assertIncorrect('python.args{a=1}, python.args{a=2}', regex='multiple values')
+
+class PythonArgumentsInLuaMethodsTest(PythonArgumentsInLuaTest):
+
+    def __init__(self, *args, **kwargs):
+        super(PythonArgumentsInLuaTest, self).__init__(*args, **kwargs)
```

Comment:
This is the kind of unexpected hack that requires looking twice to see it. Most people won't.

Reply:
You're right. I think we can get rid of this hack by making `get_args`, `get_kwargs` and `get_none` static methods, like you said.